### ISSUE TITLE
Clarifications on the internals of routing

### ISSUE THREAD

**Issue by whitfin** [seanmonstar/warp#133]
Hey there!

I'm approaching a project I originally intended to use Warp, and I have a quick question about how routing works since it was non-obvious for me. If I look at the example [here](https://github.com/seanmonstar/warp/blob/master/examples/routing.rs#L91-L96), it seems like each route is checked one by one when looking for a match. Is this the case, or is there a little more to it than I'm seeing at a glance?

Thanks in advance for any information! I just got a little curious, and I'm wondering how it might impact projects with larger numbers of routes.

**Comment by seanmonstar** [comment]
Yes, each filter is checked in order that it is added. It is possible that some apps with a large number of paths with literals may benefit from a lookup table instead. That said, you can control which paths are checked first by adding those filters earlier in the chain.
